sklearn/utils/_attribute_validation.py

```diff
@@ -0,0 +1,22 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+from dataclasses import dataclass, field
+
+
+@dataclass
```

Comment:
For my education, why use a dataclass instead of a plain class that implements `__get__` and a constructor? Taste?

Reply:
I guess the reason is NOT to have to implement the constructor 😅 Syntactic sugars are there to reduce boilerplate, that's a good enough of a reason for me to use them.